### DISCUSSION TITLE
Global Appmenu Widget support for Linux KDE Plasma top panel.

### DISCUSSION THREAD

**Discussion by sharshgargh** [Floorp-Projects/Floorp#221] [Feature Request]
Hello, Floorp has been one of the best firefox forks i have used kudos to you guys for coming up with all rich features. I use it on my Linux KDE Plasma desktop, apart from Floorp i use Waterfox fork by hawkeye116477 (github) and one feature i love about it is the support for Global Appmenu Widget on top panel of KDE Plasma, Can that be included in Floorp as well ? its very efficient way of using application menu, i have attached the screenshot of same and link to github repo.

https://github.com/hawkeye116477/waterfox-deb-rpm-arch-AppImage

![Screenshot_20230315_121452](https://user-images.githubusercontent.com/125945377/225229543-0ec3ceea-8488-4720-a73d-dfa2d225b3ec.png)

![Screenshot_20230315_122835](https://user-images.githubusercontent.com/125945377/225231079-a27544e5-4d0b-4639-88e2-93b3a0c5cdf9.png)

**Comment by Ryosuke-Asano** [comment]
We do not build app image. Do you know how to build it?

**Comment by black7375** [comment]
As far as I know, this is like unity menu support.
- https://forum.manjaro.org/t/howto-enabling-kde-global-menu-support-in-mozilla-firefox-and-thunderbird/7405
- https://github.com/WaterfoxCo/Waterfox/commit/62636cb2fb10d10fa58d8357b6634ce43fb1f5ea
- https://github.com/torvic9/plasmafox/commit/368f4b49fc64b5e9f32865c20a3b97b3a7c0d12e
- https://github.com/openSUSE/firefox-maintenance/blob/master/mozilla-kde.patch
- https://gitlab.com/dr460nf1r3/common/-/tree/master/patches/kde-upstream
- https://gitlab.com/Thaodan/firefox-kde-opensuse/-/blob/master/unity-menubar.patch

**Reply by Ryosuke-Asano** [reply]
I'll check it

**Reply by black7375** [reply]
If you have a problem in the sandbox, you can check the next patch.
https://github.com/dr460nf1r3/firedragon-browser/blob/master/firejail-fix.hook

**Reply by rusty-snake** [reply]
Regarding this "firejail-fix", from a security PoV it would be better to be more selective than allow the system-bus for all firefox based programs.

The first thing would be to allow only the D-Bus names required by appmenu.

```diff
-dbus-system none
+dbus-system filter
+dbus-system.talk org.kde.AppMenu <-- IDK the correct name, just a placeholder
```

The second thing would be to modify only firedragon.profile. At best by createing a `firedragon.local` like

```
$ cat >/etc/firejail/firedragon.local <<EOF
ignore dbus-system none
dbus-system filter
dbus-system.talk org.kde.AppMenu
EOF
```

Or you can send this upstream to the firedragon.profile if you want it to be the default for firedragon (or with an opt-in commented section in the profile).

**Comment by Ryosuke-Asano** [comment]
I researched this feature.

Adding this feature seems to disable the menubar in distributions that don't have the Unity menubar.

**Reply by dr460nf1r3** [reply]
Isn't it possible to manually re-enable it? I'm currently checking out Floorp for the first time in quite a while (did so when it was in the early stages) and this one feature is one I'd highly appreciate having around. 

How about a toggle? I realize this is a source code patch, but since I'm seeing a lot of features to be turned on and off already, it might be possible.

**Reply by Ryosuke-Asano** [reply]
I cannot do that.

You can send us a patch. 💕
